engine/dlib/src/dlib/sslsocket.cpp

```diff
@@ -194,6 +192,22 @@ Result Finalize()
     return RESULT_OK;
 }
 
+Result LoadPublicKeys(const uint8_t* key, uint32_t keylen)
+{
+    // The size of buf, including the terminating \c NULL byte in case of PEM encoded data.
+    int ret = mbedtls_x509_crt_parse(&g_SSLSocketContext.m_x509CertChain, key, keylen + 1);
+    if (ret != 0)
+    {
+        char buffer[512] = "";
+        mbedtls_strerror(ret, buffer, sizeof(buffer));
+        dmLogError("SSLSocket mbedtls_x509_crt_parse: %s0x%04x - %s", ret < 0 ? "-":"", ret < 0 ? -ret:ret, buffer);
+        return RESULT_SSL_INIT_FAILED;
+    }
+    g_SSLSocketContext.m_KeyLoaded = true;
+    mbedtls_ssl_conf_authmode( &g_SSLSocketContext.m_MbedConf, MBEDTLS_SSL_VERIFY_REQUIRED );
```

Comment:
When I see a function named "Load", I expect it to return something.
And I dont' expect a Load function so set a state in some other context (m_MbedConf). Remember, we don't _want_ global variables (g_SSLSocketContext), and there's no reason the functions should use it if they can avoid it.

I think the m_x509CertChain should be passed into the function.
And, the the `mbedtls_ssl_conf_authmode` should be done outside of the Load function.

Reply:
I agree that this is a really bad name, I replaced it with `SetSslPublicKeys` because here we override settings we made in `dmSSLSocket ::Initialize` function (because we don't have enough info when we call `dmSSLSocket ::Initialize`). IF we can parse provided buffer as ssl keys, we use it for all connections and set `MBEDTLS_SSL_VERIFY_REQUIRED`. In current implementation buffer parse doesn't make sense without MBEDTLS_SSL_VERIFY_REQUIRED and vice versa. In the future if we decide to have per-request option we can change it.

I save certificate in `g_SSLSocketContext` because it's something we gonna use for every connection (every time we use `dmSSLSocket::New`), but I set buffer not `m_x509CertChain` just to avoid using `mbedtls` funtions outside `sslsockets`, that's why in `engine.cpp` we choose path and read file as a buffer and then ask `dmSSLSocket` to set it as global keychain.